indexing-service/src/main/java/io/druid/indexing/common/actions/SegmentInsertAction.java

```diff
@@ -40,17 +41,29 @@
  * that the task cannot actually complete. Callers should avoid this by avoiding inserting too many segments in the
  * same action.
  */
-public class SegmentInsertAction implements TaskAction<Set<DataSegment>>
+public class SegmentInsertAction implements TaskAction<SegmentPublishResult>
 {
-  @JsonIgnore
   private final Set<DataSegment> segments;
+  private final DataSourceMetadata startMetadata;
+  private final DataSourceMetadata endMetadata;
+
+  public SegmentInsertAction(
+      Set<DataSegment> segments
+  )
+  {
+    this(segments, null, null);
+  }
 
   @JsonCreator
   public SegmentInsertAction(
-      @JsonProperty("segments") Set<DataSegment> segments
+      @JsonProperty("segments") Set<DataSegment> segments,
+      @JsonProperty("startMetadata") DataSourceMetadata startMetadata,
+      @JsonProperty("endMetadata") DataSourceMetadata endMetadata
```

Comment:
+1, serialization loses startMetadata/endMetadata and entries are not written into druid_dataSource table.